operations/observability/Observability.md

```diff
@@ -0,0 +1,318 @@
+
+Feb 08 23:06:38 redbull-pit grafana[82549]: logger=sqlstore t=2023-02-08T23:06:38.23+0000 lvl=info msg="Connecting to DB" dbtype=sqlite3
+Feb 08 23:06:38 redbull-pit grafana[82549]: logger=migrator t=2023-02-08T23:06:38.25+0000 lvl=info msg="Starting DB migrations"
+Feb 08 23:06:38 redbull-pit grafana[82549]: logger=migrator t=2023-02-08T23:06:38.26+0000 lvl=info msg="migrations completed" performed=0 skipped=393 durati>
+Feb 08 23:06:38 redbull-pit grafana[82549]: logger=plugin.manager t=2023-02-08T23:06:38.42+0000 lvl=info msg="Plugin registered" pluginId=input
+Feb 08 23:06:38 redbull-pit grafana[82549]: logger=query_data t=2023-02-08T23:06:38.43+0000 lvl=info msg="Query Service initialization"
+Feb 08 23:06:38 redbull-pit grafana[82549]: logger=live.push_http t=2023-02-08T23:06:38.44+0000 lvl=info msg="Live Push Gateway initialization"
+Feb 08 23:06:38 redbull-pit grafana[82549]: logger=grafanaStorageLogger t=2023-02-08T23:06:38.54+0000 lvl=info msg="storage starting"
+Feb 08 23:06:38 redbull-pit grafana[82549]: logger=ngalert t=2023-02-08T23:06:38.54+0000 lvl=info msg="warming cache for startup"
+Feb 08 23:06:38 redbull-pit grafana[82549]: logger=ngalert.multiorg.alertmanager t=2023-02-08T23:06:38.54+0000 lvl=info msg="starting MultiOrg Alertmanager"
+Feb 08 23:06:38 redbull-pit grafana[82549]: logger=http.server t=2023-02-08T23:06:38.55+0000 lvl=info msg="HTTP Server Listen" address=[::]:3000 protocol=http>
+```
+
+## IUF timing dashboard
+
+### Timing dashboard features
+
+- Use Argo workflows to collect the install/upgrade timing details.
+- Create Argo workflows to get metrics from the details available.
+- Generate Prometheus install/upgrade timing metrics.
+- Create Grafana dashboard using Prometheus metrics.
+
+### Prometheus metrics using Argo workflow
+
+The start time and end time can be obtained as Prometheus labels for the operations using Argo metrics implementation. The time stamp is recorded at the beginning and end of each operation.
+
+Metrics that are captured for the operations:
+
+- start time
+- end time
+- duration
+- status
+- product name
+- product version
+
+Metrics that are captured for the stage:
+
+- stage name
+- stage type
+- stage start time (the earliest start time of the all the operations in the stage)
+- stage end time (the latest end time of the all the operations in the stage)
+- stage duration (the difference between the stage end time and stage start time)
+- stage status (status is marked as succeeded if all the operations' statuses are succeeded and vice versa)
+
+Metrics that are captured for the product:
+
+- product name
+- product start time (start time of the process-media stage)
+- product end time (start time of the post-install-check stage)
+- product status (status is marked as succeeded if all the stages' statuses are succeeded and vice versa)
```

Comment:
Is this accurate?

```suggestion
- product status (status is marked as succeeded if all the stages' statuses are succeeded, otherwise it is marked as failed)
```

Reply:
changed.